Does h5cpp work with libhdf5 1.10.5 ?

Hi,

If anyone has tried to build h5cpp with libhdf5 1.10.5?

My college Vijay has tried to do it and he has got:

```h5cpp/src/h5cpp/file/direct_driver.cpp: In member function ‘virtual void hdf5::file::DirectDriver::operator()(const hdf5::property::FileAccessList&) const’:
h5cpp/src/h5cpp/file/direct_driver.cpp:80:48: error: invalid static_cast from type ‘const hdf5::property::FileAccessList’ to type ‘hid_t {aka long int}’
   if(H5Pset_fapl_direct(static_cast<hid_t>(fapl),alignment_,
                                                ^
make[2]: *** [src/h5cpp/CMakeFiles/h5cpp.dir/build.make:453: src/h5cpp/CMakeFiles/h5cpp.dir/file/direct_driver.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:203: src/h5cpp/CMakeFiles/h5cpp.dir/all] Error 2
make: *** [Makefile:130: all] Error 2```

--- a/src/h5cpp/file/direct_driver.cpp
+++ b/src/h5cpp/file/direct_driver.cpp
@@ -25,6 +25,7 @@
 
 #include <h5cpp/core/hdf5_capi.hpp>
 #include <h5cpp/file/direct_driver.hpp>
+#include <h5cpp/property/file_access.hpp>
 #include <h5cpp/error/error.hpp>
 
 #ifdef H5_HAVE_DIRECT

--- a/test/file/driver_test.cpp
+++ b/test/file/driver_test.cpp
@@ -37,6 +37,14 @@ TEST(MemoryDriver, create_default)
   EXPECT_EQ(m.id(), file::DriverID::eMemory);
 }
 
+#ifdef H5_HAVE_DIRECT
+TEST(DirectDriver, create_default)
+{
+  file::DirectDriver m;
+  EXPECT_EQ(m.id(), file::DriverID::eDirect);
+}
+#endif
+
 TEST(MemoryDriver, create_custom)
 {
   file::MemoryDriver m(7000, true);

